realm/realm-annotations-processor/src/test/resources/io/realm/AllTypesRealmProxy.java

```diff
@@ -599,10 +583,28 @@ public static AllTypes copyOrUpdate(Realm realm, AllTypes object, boolean update
         if (canUpdate) {
             Table table = realm.getTable(AllTypes.class);
             long pkColumnIndex = table.getPrimaryKey();
-            if (object.getColumnString() == null) {
+            String primaryKey;
+            if (object.realm != null) {
+                primaryKey = ((AllTypesRealmProxy)object).realmGetter$columnString();
+            } else {
+                Field field = null;;
+                Class<? extends AllTypes> clazz = object.getClass();
+                try {
+                    field = clazz.getDeclaredField("columnString");
```

Comment:
IMO this is definitely something that should be cached, but it is fine to add it to a TODO list until after this PR is merged.